sonar-plugin-api/src/main/java/org/sonar/api/batch/AbstractCpdMapping.java

```diff
@@ -19,15 +19,16 @@
  */
 package org.sonar.api.batch;
 
-import org.sonar.api.resources.Resource;
-
 import java.util.List;
+import org.sonar.api.resources.Resource;
 
 /**
  * A pre-implementation of the CpdMapping extension point
  *
  * @since 1.10
+ * @deprecated since 5.6 use {@link SensorContext#newCpdTokens()}
  */
+@Deprecated
 public abstract class AbstractCpdMapping implements CpdMapping {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1133)